lib/person.rb

```diff
@@ -7,12 +9,17 @@ def initialize(age, name: 'Unknow', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @corrector = Corrector.new
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
+  def validate_name
+    @corrector.correct_name(@name)
```

Comment:
So after `validate_name` changes the instance `@name` where will be the new name stored? 

In other words.. the returned value from `@corrector.correct_name(@name)`  where will be stored?

```suggestion
    @name = @corrector.correct_name(@name)
```